examples/implementation_languages/coffeescript/coffeescript_json.coffee

```diff
@@ -0,0 +1,97 @@
+chevrotain = require 'chevrotain'
+Token = chevrotain.Token
+Lexer = chevrotain.Lexer
+Parser = chevrotain.Parser
+
+class True extends Token
+  @PATTERN: /true/
+
+class False extends Token
+  @PATTERN: /false/
+
+class Null extends Token
+  @PATTERN: /null/
+
+class LCurly extends Token
+  @PATTERN: /{/
+
+class RCurly extends Token
+  @PATTERN: /}/
+
+class LSquare extends Token
+  @PATTERN: /\[/
+
+class RSquare extends Token
+  @PATTERN: /]/
+
+class Comma extends Token
+  @PATTERN: /,/
+
+class Colon extends Token
+  @PATTERN: /:/
+
+class StringLiteral extends Token
+  @PATTERN: /"(:?[^\\"]+|\\(:?[bfnrtv"\\/]|u[0-9a-fA-F]{4}))*"/
+
+class NumberLiteral extends Token
+  @PATTERN: /-?(0|[1-9]\d*)(\.\d+)?([eE][+-]?\d+)?/
+
+class WhiteSpace extends Token
+  @PATTERN = /\s+/
+  @GROUP = Lexer.SKIPPED
+
+allTokens = [WhiteSpace, NumberLiteral, StringLiteral, LCurly, RCurly, LSquare, RSquare, Comma, Colon, True, False, Null]
+JsonLexer = new Lexer allTokens
+
+class JsonParserCoffeeScript extends Parser
+  constructor: (input) ->
+    super input, allTokens
+
+    @RULE "json", =>
+      @.OR [
+        {ALT: => @.SUBRULE @.object}
+        {ALT: => @.SUBRULE @.array}
+      ]
+
+    @RULE "object", =>
+      @.CONSUME LCurly
+      @.MANY_SEP Comma, =>
+        @.SUBRULE2 @.objectItem
+      @.CONSUME RCurly
+
+    @RULE "objectItem", =>
+      @.CONSUME StringLiteral
```

Comment:
Thanks for the de-structuring tip 👍 fixed it.

> I see the revised version of the RULE() function. That's nice. Thank you.

Realized something [better](https://github.com/witheve/Eve/blob/8de09b41fd7f01b16a33190b42732e74afee9f6e/src/runtime/parser.ts#L402) can/should be done thanks to the Eve's parser and the previous issue you opened regarding the APIs.

CoffeeScript is really terse and easy on the eyes for reading.
I definitively see why a less verbose syntax (as you suggested) would be more idiomatic  for defining the Tokens. the text area which defines the Tokens feels a little out of place in CoffeeScript...